### DISCUSSION TITLE
Write request without StatusCode and SourceTimeStamp

### DISCUSSION THREAD

**Discussion by NoHeinz** [FreeOpcUa/opcua-asyncio#720] [Q&A] [answered]
I am currently trying to communicate with a Siemens S7-1500 PLC and they have their own OPC server that I believe doesn't allow the StatusCode and SourceTimeStamp in their packets for write requests(see pictures attached). Read request from python work fine though. Is there anyway to exclude those values in a write request? The prosys client that I am using sends the packets correctly and gets back a successful packet.(see picture attached)
P.S. I'm using a synchronous client but I used the asynchronous client also and it has the same problems
**PYTHON CODE**
![python_code](https://user-images.githubusercontent.com/61035266/141539637-afb18247-8e9f-4016-aac3-a398b9b6283f.png)

**WRITE REQUEST PACKET FROM PYTHON TO PLC**
![python_write_request](https://user-images.githubusercontent.com/61035266/141539640-5f03c2c0-fb3d-4897-82b1-25138144301f.png)

**WRITE RESPONSE TO PYTHON PACKET FROM PLC**
![PLC_response_python](https://user-images.githubusercontent.com/61035266/141539641-631936a1-b042-477a-a70e-030a295e86d8.png)

**ERROR TEXT**
![image](https://user-images.githubusercontent.com/61035266/141803409-782d21ba-6107-49f7-b9ca-27975886d743.png)

**WRITE REQUEST PACKET FROM PROSYS OPC CLIENT(works)**
![prosys_write_request](https://user-images.githubusercontent.com/61035266/141539642-48819b6b-0092-4670-b53e-2bf1f8db6006.png)
?

**Comment by NoHeinz** [comment, marked as answer]
The siemens opc ua server doesn't accept write requests with SourceTimestamp in the Value:DataValue section, I had to change the way it builds the datavalue of the write request in the python package itself in common/ua-utils.py in the value_to_datavalue function: 
![image](https://user-images.githubusercontent.com/61035266/142017207-f3437cac-adb2-49bc-ad5e-82769a127389.png)
just by deleting the parameter of SourceTimestamp when instantiating the DataValue for write requests I was successfully writing to the PLC's OPC server. I wish there was a better way to do this but it works for now and I dont believe this messes anything else up since the packet already has a timestamp in the header.

**Reply by AndreasHeine** [reply]
you could pass a propper datavalue to the write method rather than a value!
working with siemens alot, so if you call write_value(True) some typeguessing and convinience functions com into play (which can lead to your problems...)

we had that already in multiple issues and discussions resolved!

**Reply by AndreasHeine** [reply]
even if its not asyncua the logic is almost the same:
https://github.com/FreeOpcUa/python-opcua/issues/1159
https://github.com/FreeOpcUa/python-opcua/issues/1069
